markdown/developers/adding_modules.md

```diff
@@ -0,0 +1,406 @@
+---
+title: Adding a new module
+subtitle: Follow this walkthrough for adding and reviewing new nf-core modules
+---
+
+## Adding a new module file
+
+If you decide to upload a module to `nf-core/modules` then this will ensure that it will become available to all nf-core pipelines, and to everyone within the Nextflow community! See [`modules/`](modules)
+for examples.
+
+## Terminology
+
+The features offered by Nextflow DSL2 can be used in various ways depending on the granularity with which you would like to write pipelines. Please see the listing below for the hierarchy and associated terminology we have decided to use when referring to DSL2 components:
+
+- *Module*: A `process` that can be used within different pipelines and is as atomic as possible i.e. cannot be split into another module. An example of this would be a module file containing the process definition for a single tool such as `FastQC`. At present, this repository has been created to only host atomic module files that should be added to the [`modules/`](modules/) directory along with the required documentation and tests.
```

Comment:
Could we make these have their own level 3 headings please? Then we can link to them directly.